IdentifiabilityODE.mpl

```diff
@@ -342,10 +365,17 @@ IdentifiabilityODE := proc(system_ODEs, params_to_assess, {p := 0.99, count_solu
     end do:    
   elif method = 2 then
     gb := Groebner[Basis]([op(Et_hat), z_aux * Q_hat - 1], tdeg(op(vars)));
+    
     for i from 1 to nops(theta_l) do
-      if Groebner[NormalForm](theta_l[i], gb, tdeg(op(vars))) = subs(theta_hat, theta_l[i]) then
-        theta_g := [ op(theta_g), theta_l[i] ]:
-      end if:
+      if use_weights then
+       	if Groebner[NormalForm](theta_l[i]^degree(weights_table[theta_l[i]]), gb, tdeg(op(vars)), characteristic=char) = check then
+         		theta_g := [op(theta_g), theta_l[i]]:
+       	end if:
+       else
+       	if Groebner[NormalForm](theta_l[i], gb, tdeg(op(vars)), characteristic=char) = check then
+         		theta_g := [op(theta_g), theta_l[i]]:
+       	end if:

```

Comment:
We could avoid this case distinction if we always create `weights_table` but in the no-weights case put all ones there.